lib/spack/spack/environment/environment.py

```diff
@@ -467,6 +468,9 @@ def from_dict(base_path, d):
 
     @property
     def _current_root(self):
```

Comment:
Can this have a doctoring?  Reading this I don't know what it's *supposed* to return.  It seems like it returns `None` for non-renameat views when there isn't  a symlink... but that's going to confuse things with the new implementation. 

Consider:
1. user does a directory view with `renameat2`
2. switches to symlinks
3. this will now say there is no current root, but that's probably wrong as there is a whole directory there.

And, yes, we don't have a config option for this, but:
* We could, and we might want to
* People check out different versions of Spack all the time.
* You might have a view exposed to different machines, only one of which supports `renameat2`

I think for that last reason, this needs to be a config option for views that gets stored *in* the view directory -- i.e. a view identifies itself as a `renameat2` view and says it needs to be updated a certain way.

Reply:
this still needs a docstring